test/micro/org/openjdk/bench/jdk/incubator/vector/bigdata/SelectiveStore.java

```diff
@@ -0,0 +1,189 @@
+        tld.conflict_scalar[tld.conflict_scalar_cnt++] = tld.index[i];
+      }
+    }
+  }
+
+  @Benchmark
+  @Threads(8)
+  public void selectiveStore_vector_128(ThreadLocalData tld) {
+    tld.conflict_vector_cnt = 0;
+    for (int i = 0; i < ARRAY_LENGTH; i += INT_128_SPECIES.length()) {
+      IntVector av = IntVector.fromArray(INT_128_SPECIES, tld.input1, i);
+      IntVector bv = IntVector.fromArray(INT_128_SPECIES, tld.input2, i);
+      IntVector cv = IntVector.fromArray(INT_128_SPECIES, tld.index, i);
+      VectorMask<Integer> mask = av.compare(VectorOperators.NE, bv);
+      IntVector dv = cv.compress(mask);
+      dv.intoArray(tld.conflict_vector, tld.conflict_vector_cnt, mask.prefixMask());
+      tld.conflict_vector_cnt += mask.trueCount();
+    }
+  }
+
+  @Benchmark
+  @Threads(8)
+  public void selectiveStore_vector_256(ThreadLocalData tld) {
+    tld.conflict_vector_cnt = 0;
+    for (int i = 0; i < ARRAY_LENGTH; i += INT_256_SPECIES.length()) {
+      IntVector av = IntVector.fromArray(INT_256_SPECIES, tld.input1, i);
+      IntVector bv = IntVector.fromArray(INT_256_SPECIES, tld.input2, i);
+      IntVector cv = IntVector.fromArray(INT_256_SPECIES, tld.index, i);
+      VectorMask<Integer> mask = av.compare(VectorOperators.NE, bv);
+      IntVector dv = cv.compress(mask);
+      dv.intoArray(tld.conflict_vector, tld.conflict_vector_cnt, mask.prefixMask());
+      tld.conflict_vector_cnt += mask.trueCount();
+    }
+  }
+
+  @Benchmark
+  @Threads(8)
+  public void selectiveStore_vector_512(ThreadLocalData tld) {
+    tld.conflict_vector_cnt = 0;
+    for (int i = 0; i < ARRAY_LENGTH; i += INT_512_SPECIES.length()) {
+      IntVector av = IntVector.fromArray(INT_512_SPECIES, tld.input1, i);
+      IntVector bv = IntVector.fromArray(INT_512_SPECIES, tld.input2, i);
+      IntVector cv = IntVector.fromArray(INT_512_SPECIES, tld.index, i);
+      VectorMask<Integer> mask = av.compare(VectorOperators.NE, bv);
+      IntVector dv = cv.compress(mask);
+      dv.intoArray(tld.conflict_vector, tld.conflict_vector_cnt, mask.prefixMask());
+      tld.conflict_vector_cnt += mask.trueCount();
+    }
+  }
+
+  @Benchmark
+  @Threads(8)
+  public void selectiveStore_vector_preferred(ThreadLocalData tld) {
+    tld.conflict_vector_cnt = 0;
+    for (int i = 0; i < ARRAY_LENGTH; i += INT_PREFERRED_SPECIES.length()) {
+      IntVector av = IntVector.fromArray(INT_PREFERRED_SPECIES, tld.input1, i);
+      IntVector bv = IntVector.fromArray(INT_PREFERRED_SPECIES, tld.input2, i);
+      IntVector cv = IntVector.fromArray(INT_PREFERRED_SPECIES, tld.index, i);
+      VectorMask<Integer> mask = av.compare(VectorOperators.NE, bv);
+      IntVector dv = cv.compress(mask);
+      dv.intoArray(tld.conflict_vector, tld.conflict_vector_cnt, mask.prefixMask());
+      tld.conflict_vector_cnt += mask.trueCount();
+    }
+  }
+
+  @Benchmark
+  @Threads(8)
+  public void selectiveStore_verify(ThreadLocalData tld) {
```

Comment:
Is it necessary to benchmark the correctness?

Reply:
Thanks Eric for your review :-)
This function was imported to ensure compress API's correctness during my development process.
I will remove this function as your suggestion after all dependent patches are merged.